### PR TITLE
REF: implement array_algos.putmask

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/array_algos/putmask.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+from pandas._libs import lib
+from pandas._typing import ArrayLike
+
+from pandas.core.dtypes.cast import convert_scalar_for_putitemlike, maybe_promote
+from pandas.core.dtypes.common import is_float_dtype, is_integer_dtype, is_list_like
+from pandas.core.dtypes.missing import isna_compat
+
+
+def putmask_simple(values: ArrayLike, mask: np.ndarray, value: Any):
```
Comment (jreback) [inline pandas/core/array_algos/putmask.py:17]: returns -> None

Comment (jreback) [reply pandas/core/array_algos/putmask.py:17]: though *maybe* we should return the array for consistency with putmask_simple?

Comment (jreback) [reply pandas/core/array_algos/putmask.py:17]: or rename to putmask_inplace

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/array_algos/putmask.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/array_algos/putmask.py:17]: > though maybe we should return the array for consistency with putmask_simple?

i think not returning anything makes it more obvious that it is inplace

> or rename to putmask_inplace

sure

**Review by jreback — APPROVED** [review]

**Pull request merged**